test/gas/gas.js

```diff
@@ -0,0 +1,563 @@
+const assert = require("assert");
+const sleep = require("../helpers/utils/sleep");
+const bootstrap = require("../helpers/contract/bootstrap");
+const getWeb3 = require("../helpers/web3/getWeb3");
+const isGasExpenseCorrect = require("./lib/isGasExpenseCorrect");
+const testTransactionEstimate = require("./lib/transactionEstimate");
+const toBytes = require("./lib/toBytes");
+const deployContract = require("./lib/customContractDeploy");
+
+const RSCLEAR_REFUND = 15000;
+const RSCLEAR_REFUND_FOR_RESETTING_DIRTY_SLOT_TO_ZERO = 19800;
+const RSELFDESTRUCT_REFUND = 24000;
+
+// Thanks solc. At least this works!
+// This removes solc's overzealous uncaughtException event handler.
+process.removeAllListeners("uncaughtException");
+
+describe("Gas", () => {
+  const mainContract = "EstimateGas";
+  const contractFilenames = [];
+  const contractPath = "../../contracts/gas/";
+
+  const mnemonic = "candy maple cake sugar pudding cream honey rich smooth crumble sweet treat";
+  const options = { mnemonic };
+  const services = bootstrap(mainContract, contractFilenames, options, contractPath);
+
+  let deploymentReceipt;
+
+  before("Adding funds to the contract", async function() {
+    this.timeout(10000);
+    const { accounts, bytecode, contract } = services;
+
+    const promiEvent = contract.deploy({ data: bytecode }).send({
```

Comment:
why do we call `contract.deploy` here? shouldn't we improve `bootstrap`, `compileAndDeploy`, and whatever else to make it possible to transfer eth on the deploy?

Reply:
`deploy` now exports the `receipt`.